explainer/Explainer11.md

```diff
@@ -0,0 +1,287 @@
+
+In this context, the MyLampThing example above would result in declaring status a property, toggling an action, and overheating an event. 
+
+Now, by reading the above example TD,  one can obtain knowledge about the Thing named `MyLampThing`, including the following:
+
+- The Thing requires the use of HTTP Basic Authentication when 
+  accessing all interaction resources (announced by securityDefinitions and security).
+- The Thing provides one Property interaction resource with the name `status`.
+  - The property `status` is accessible via the HTTP protocol
+    with a GET method at URI `https://192.168.2.134/status`.
+  - Reading the property `status` will return a string value.
+- The Thing provides one Action interaction resource with the name `toggle`. 
+  - The action `toggle` is accessible via the HTTP protocol
+    with a POST method at URI `https://192.168.2.134/toggle`.
+- The Thing provides one Event interaction resource with the name `overheating`.
+  - The event `overheating` can be obtained at URI 
+    `https://192.168.2.134/oh` by using HTTP with a long polling subprotocol.
+    The subprotocol clarifies which of several possible mechanisms in HTTP
+    are used to provide "push" notifications.
+  - Each message pushed by the Thing is a string value.
+
+
+## What you can do with a Thing Description (TD)?
+
+Based on such information in a TD a WoT stack such as designed based on the [W3C Web of Things Scripting API](https://w3c.github.io/wot-scripting-api/) can simple interpret the TD content and abstract all the communication details. Developer just working with the **properties, actions**, and **events** paradigm and do not need to take care protocol specifics such as the HTTP method, the resource address, port number, etc.. The following video gives an inside about this aspect:
+
+[![IMAGE ALT TEXT](http://img.youtube.com/vi/lt_P2BU8e3I/0.jpg)](http://www.youtube.com/watch?v=lt_P2BU8e3I "TD usage with a programming API")
+
+This example also shows that TDs can be used to easily onboard Things into any IoT ecosystems (e.g., IoT cloud services, edge systems, etc.). TDs helps to understand what the Things offers and how the offered data and functions can be bound to applications processes (e.g., creating service mash-ups, dashboards, etc.). The following example shows how TD are used in a node-red application:
+
+[![IMAGE ALT TEXT](http://img.youtube.com/vi/oAcYbJ6P9bU/0.jpg)](http://www.youtube.com/watch?v=oAcYbJ6P9bU "TD usage in node-red")
+
+TD also helps to mange your IoT system that typically consist of a heterogeneous device landscape based on different vendors and technologies that are used (e.g., HTTP, MQTT, Modbus, etc). Directories can be used to manage the TDs (e.g., create, search, etc). [The W3C WoT Discovery](https://w3c.github.io/wot-discovery/) provides details about a standardized how the interface of such directories can look like.  
```

Comment:
... manage IoT systems.  An IoT system typically consists of a heterogenous set of devices provided by different vendors and based on different technologies and protocols (e.g. HTTP, MQTT, Modbus, etc).

Reply:
This paragraph is also the first time that Discovery is mentioned, but the purpose of Discovery is distribution, not management of TDs.  In particular directories are designed for transient storage and include features like expiry.  They aren't the best choice for long-term archival.

If you are going to introduce Discovery here, I would say something like "The WoT Discovery specification defines mechanisms for controlled distribution and access to TDs, including searchable directories in which TDs can be dynamically registered."

Reply:
done